consul/src/main/java/com/linecorp/armeria/server/consul/ConsulUpdatingListenerBuilder.java

```diff
@@ -0,0 +1,118 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.consul;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.internal.consul.ConsulClient;
+
+/**
+ * Builds a new {@link ConsulUpdatingListener}, which registers the server to a Consul cluster.
+ * <h2>Examples</h2>
+ * <pre>{@code
+ * ConsulUpdatingListener listener =
+ *     new ConsulUpdatingListenerBuilder("myService").url("http://myConsulHost:8500").build();
+ * ServerBuilder sb = Server.builder();
+ * sb.addListener(listener);
+ * }</pre>
+ */
+public final class ConsulUpdatingListenerBuilder {
+    private final String serviceName;
+    @Nullable private String consulUrl;
+    @Nullable private Endpoint serviceEndpoint;
+
+    @Nullable private String checkUrl;
+    @Nullable private String checkMethod;
+    @Nullable private String checkInterval;
+
+    /**
+     * Creates a {@link ConsulUpdatingListenerBuilder} with a service name.
+     *
+     * @param serviceName Service name to register
+     */
+    ConsulUpdatingListenerBuilder(String serviceName) {
+        this.serviceName = requireNonNull(serviceName, "serviceName");
+        checkArgument(!this.serviceName.isEmpty(), "serviceName can't be empty");
+    }
+
+    /**
+     * Sets the Consul's URL.
+     *
+     * @param consulUrl URL of consul agent, e.g.: http://127.0.0.1:8500
+     */
+    public ConsulUpdatingListenerBuilder url(String consulUrl) {
+        this.consulUrl = requireNonNull(consulUrl, "consulUrl");
+        checkArgument(!this.consulUrl.isEmpty(), "consulUrl can't be empty");
+        return this;
+    }
+
+    /**
+     * Sets URL for checking health by consul agent.
+     *
+     * @param checkUrl URL for checking health of service
+     */
+    public ConsulUpdatingListenerBuilder checkUrl(String checkUrl) {
+        this.checkUrl = requireNonNull(checkUrl, "checkUrl");
+        return this;
+    }
+
+    /**
+     * Sets HTTP method for checking health by consul agent.
+     *
+     * @param checkMethod HTTP method for checking health of service
+     */
+    public ConsulUpdatingListenerBuilder checkMethod(String checkMethod) {
+        this.checkMethod = requireNonNull(checkMethod, "checkMethod");
+        return this;
+    }
+
+    /**
+     * Sets checkInterval.
+     * e.g.: "10s", "1m"
+     */
+    public ConsulUpdatingListenerBuilder checkInterval(String checkInterval) {
```

Comment:
Armeria uses `Duration` and milliseconds to set a specified period.
Could you replace this with the following APIs to provide consistent API?
```java
public ConsulUpdatingListenerBuilder checkInterval(long checkInteralMillis) {
    ...
}
public ConsulUpdatingListenerBuilder checkInterval(Duration checkInterval) {
    ...
}
``